orca-igor/src/test/groovy/com/netflix/spinnaker/orca/igor/tasks/GetBuildArtifactsTaskSpec.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.igor.tasks
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.igor.BuildService
+import com.netflix.spinnaker.orca.pipeline.model.Execution
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import spock.lang.Specification
+import spock.lang.Subject
+
+class GetBuildArtifactsTaskSpec extends Specification {
+  def buildService = Mock(BuildService)
+  def testArtifact = Artifact.builder().name("my-artifact").build()
+
+  def BUILD_NUMBER = 4
+  def MASTER = "builds"
+  def JOB = "orca"
+  def PROPERTY_FILE = "my-file"
+
+  @Subject
+  GetBuildArtifactsTask task = new GetBuildArtifactsTask(buildService)
+
+  def "retreives artifacts and adds them to the stage outputs"() {
+    given:
+    def stage = createStage(PROPERTY_FILE)
+
+    when:
+    TaskResult result = task.execute(stage)
+    def artifacts = result.getOutputs().get("artifacts") as List<Artifact>
+
+    then:
+    1 * buildService.getArtifacts(BUILD_NUMBER, PROPERTY_FILE, MASTER, JOB) >> [testArtifact]
+    artifacts.size() == 1
+    artifacts.get(0).getName() == "my-artifact"
+  }
+
+  def "handles an empty artifact list"() {
+    given:
+    def stage = createStage(PROPERTY_FILE)
+
+    when:
+    TaskResult result = task.execute(stage)
+    def artifacts = result.getOutputs().get("artifacts") as List<Artifact>
+
+    then:
+    1 * buildService.getArtifacts(BUILD_NUMBER, PROPERTY_FILE, MASTER, JOB) >> []
+    artifacts.size() == 0
+  }
+
+  def "does not fetch artifacts if the property file is empty"() {
```

Comment:
Why should artifacts not be fetched if the property file is empty?

Reply:
The way to specify the Jinja template to use to inflate artifacts is by passing a property in the property file, so there's nothing to do if we don't have a template.

It might make sense at some point to allow this to be configured more statically (maybe every job from the jenkins server uses a given template, or maybe the trigger defines it) but for now it's defined via an output property.

Reply:
Ahhh, I was missing that piece. Thanks for the explanation. So the whole template is in the property file, or it contains a reference to it?

Reply:
It's just a reference to it; you can either specify a built-in template by name (currently jar or jinja) or a custom template that you've configured in your `igor.yml`:
```
artifacts:
  templates:
  - name: <name of template>
    templatePath: <path to the template>
```
So basically the job itself says "this is how to extract an artifact", though I could also at some point in the future see the benfit of Spinnaker just knowing "this job always needs to get artifacts via this template".